src/main/java/org/fenixedu/academic/service/services/bolonhaManager/CreateCompetenceCourse.java

```diff
@@ -42,6 +42,26 @@
 
     @Atomic
     public static CompetenceCourse run(String name, String nameEn, String acronym, Boolean basic, RegimeType regimeType,
+            CompetenceCourseLevel competenceCourseLevel, CompetenceCourseType type, String unitID,
```

Comment:
With this change, the code in the method will be duplicated with the newly deprecated one, minus the `setCode`. You should reuse some of the code.